libbeat/outputs/elasticsearch/utils.go

```diff
@@ -0,0 +1,18 @@
+package elasticsearch
+
+import (
+	"unicode"
+	"github.com/elastic/beats/libbeat/logp"
+	"strings"
+)
+
+func TryLowercaseIndex(index string) string {
+	for _, u := range []rune(index) {
+		if unicode.IsUpper(u) {
+			newIndex := strings.ToLower(index)
+			logp.Warn("Index name %s is invalid, replace it with %s", index, newIndex)
+			return newIndex
+		}
+	}
+	return index
+}
```

Comment:
There is a few things that bug me in this:
- This could get noisy since it could be potentially be logged on every events.
- Worse case when the index is OK we iterate on all the chars, so an happy path will be hit by this.
- Due to the dynamic nature of index, we use string interpolation here, we have to do this check on every events.

I wonder if instead, we could **provide a better errors message** when this situation happen and either tell them to use a lowercase index or we could provide a lowercase processor.

Reply:
Could we provide a better errors message and a additional flag like `--lowercase-index` to enable the processor?

Reply:
Personnaly I would not add a custom flag, they are easy to add, hard to remove and it make the CLI more complicated. 

But the error message could mention how to fix the situation.

"Use the lowercase processor to normalize your index"

Reply:
@ruflin would you mind give your opinion on my [comment](https://github.com/elastic/beats/pull/6640#discussion_r177299610)?

Reply:
First of all I don't like magic. Sending a field with upper case and then converting it to lower case automatically for the index naming feels like magic to me. I see multiple options moving forward:

* A config option in the elasticsearch output to normalies / lowercase the index which is off by default.
* A lower case processor. The downside of the lower case processor is that if it's applied on a field, I expect it to be also changed in the final event it is sent.

I'm leaning towards the output config for the reason that this is an Elasticsearch specific issue so it should be an Elasticsearch config option. Having a config option off by default will also make it clear that when turning it on, it will have some processing overhead.

+1 on providing better error messages.

Reply:
> A lower case processor. The downside of the lower case processor is that if it's applied on a field, I expect it to be also changed in the final event it is sent.

Correct.

Glad we got a conclusion.

@Hexilee 

I would be +1 to add a specific options to the elasticsearch output to lowercase the index name and make it off by default.

If you move forward with the changes can you sign the [CLA](https://www.elastic.co/contributor-agreement)?

Reply:
@ph Yeah, I'd like to, and I have signed the CLA.
However, as you say, current error message is generated by ElasticSearch, so should we improve ElasticSearch only? Whatever providing better error messages or lowercase processor.